src/styles/_catalog.scss

```diff
@@ -1,22 +1,30 @@
-.catalog {
-  display: flex;
-  flex-wrap: wrap;
+@import "./utils/variables";
 
-  max-width: 944px;
-  margin: 0 auto;
+.catalog {
+  display: grid;
+  grid-template-columns: repeat(4, $card-width);
+  gap: 48px;
+  justify-content: center;
 
   &__card {
     display: flex;
     justify-content: center;
     align-items: center;
-    width: 200px;
+    width: $card-width;
     height: 300px;
     border: 1px solid #ccc;
     border-radius: 5px;
-    margin-bottom: 48px;
   }
 
-  &__card:not(:nth-child(4n)) {
-    margin-right: 48px;
+  @media (max-width: 1024px) {
+    grid-template-columns: repeat(3, $card-width);
+  }
+
+  @media (max-width: 768px) {
+    grid-template-columns: repeat(2, $card-width);
+  }
```

Comment:
Also, one more trick - you could set the SCSS variable for the number of cards and change in the media query.
Example:
```
--cards-number: 4;

.catalog {
  grid-template-columns: repeat(var(--cards-number), $card-width);
  
  @media (min-width: 488px) {
     --cards-number: 2;
  }
   /* and so on */
}
```
More info [here](https://www.w3schools.com/css/css3_variables_mediaqueries.asp)

Reply:
This is really cool, code looks a lot cleaner. 

>  you could set the SCSS variable for the number of cards  

Pretty sure its typo since your example using CSS variable, but i have to write it anyway for someone who may check this code for reference. SCSS variable can only have one value because its being converted to CSS.